src/TapeChart.js

```diff
@@ -0,0 +1,48 @@
+class TapeChart {
+  constructor(users, rooms, bookings) {
+    this.users = users;
+    this.rooms = rooms;
+    this.bookings = bookings;
+  }
+
+getAvailableRoomsByDate(date) {
+    let availableByDate = this.rooms.reduce((acc, room) => {
+      let bookingsByRoom = this.bookings.filter(booking =>
+        booking.roomNumber === room.number);
+      if (!bookingsByRoom.some(booking => booking.date === date)) {
```

Comment:
really interesting, I've never seen this. I googled it and it looked like it returns a boolean. .find is very similar but I think it can return values. 

values might help, but that is just a hunch. Don't chase after this lead for more then a minute.

Reply:
I took another look at line 12.  It's the first time I've used `some()` actually.  I did two expect statements to make sure it wasn't giving me a false positive. 

My psuedocode for this line is `if none of the bookings for the specific room number being iterated over have the date that I care about, then push that room into the list of available rooms (my acc)`.

Since I don't care which ones have that date, I only want to meet the conditional to determine whether to push the room or not, I went with some() over find().